### ISSUE TITLE
Do not hardcode SAI/ESAI parameters

### ISSUE THREAD

**Issue by dbaluta** [thesofproject/sof#2520]
@paulstelian97 I think we should not hardcode parameters but try to read them from sai->params. 

We need something similar with ssp_get_hw_params. 

_Originally posted by @dbaluta in https://github.com/thesofproject/sof/pull/2500_

**Comment by dbaluta** [comment]
@IulianOlaru249 @constantinraducanu you can have a look at this if you have some time.

**Comment by ymdatta** [comment]
@dbaluta: Yes, i will take a look at this. Can you assign this issue to me? Thanks.

**Comment by costi2** [comment]
@dbaluta I will look on this too.

**Comment by dbaluta** [comment]
@ymdatta @costi2 @IulianOlaru249 great o know about your interest. 

If you have time you can also try to see if you can tackle one of unassigned issues from here. Check with the reporter of the issue for more details.

https://github.com/thesofproject/sof/issues?q=is%3Aissue+is%3Aopen+FEATURE
